src/materialized/tests/auth.rs

```diff
@@ -1345,3 +1364,83 @@ fn test_auth() -> Result<(), Box<dyn Error>> {
 
     Ok(())
 }
+
+#[test]
+fn test_auth_intermediate_ca() -> Result<(), Box<dyn Error>> {
+    // Create a CA, an intermediate CA, and a server key pair signed by the
+    // intermediate CA.
+    let ca = Ca::new_root("test ca")?;
+    let intermediate_ca = ca.request_ca("intermediary")?;
+    let (server_cert, server_key) =
+        intermediate_ca.request_cert("server", vec![IpAddr::V4(Ipv4Addr::LOCALHOST)])?;
+
+    // Create a certificate chain bundle that contains the server's certificate
+    // and the intermediate CA's certificate.
+    let server_cert_chain = {
+        let path = intermediate_ca.dir.path().join("server.chain.crt");
+        let mut buf = vec![];
+        File::open(&server_cert)?.read_to_end(&mut buf)?;
+        File::open(intermediate_ca.ca_cert_path())?.read_to_end(&mut buf)?;
+        fs::write(&path, buf)?;
+        path
+    };
+
+    // When the server presents only its own certificate, without the
+    // intermediary, the client should fail to verify the chain.
+    let config = util::Config::default().with_tls(TlsMode::Require, &server_cert, &server_key);
+    let server = util::start_server(config)?;
+    run_tests(
+        "TlsMode::Require",
+        &server,
+        &[
+            TestCase::Pgwire {
+                user: "materialize",
+                password: None,
+                ssl_mode: SslMode::Require,
```

Comment:
wait, is this the right SSL mode? I thought Require only ensures that SSL is in use...

Reply:
Yeah, this is the _server_ SSL mode, not the client SSL mode! It's confusing. But `materialized --tls-mode=require` is what you want if you want to force the client to use SSL, but not request a client cert. The client does verify the server's cert in this test, by virtue of not saying `b.set_verify(SslVerifyMode::NONE)` in the `configure` callback.

Reply:
Oh man, yeah, now that you explain it, that makes sense. Ok, looks great!